tutorials/math/random_number_generation.rst

```diff
@@ -409,17 +443,54 @@ ends up empty. When that happens, you reinitialize it to its default value::
 
 
     func get_fruit():
-        if _fruits.empty():
+        if _fruits.is_empty():
             # Fill the fruits array again and shuffle it.
             _fruits = _fruits_full.duplicate()
             _fruits.shuffle()
 
         # Get a random fruit, since we shuffled the array,
         # and remove it from the `_fruits` array.
         var random_fruit = _fruits.pop_front()
-        # Prints "apple", "orange", "pear", or "banana" every time the code runs.
+        # Returns "apple", "orange", "pear", or "banana" every time the code runs, removing it from the array.
+        # When all fruit are removed, it refills the array.
         return random_fruit
 
+ .. code-tab:: csharp
+
+    private Godot.Collections.Array _fruits = new Godot.Collections.Array{ "apple", "orange", "pear", "banana" };
+    // A copy of the fruits array so we can restore the original value into `fruits`.
+    private Godot.Collections.Array _fruitsFull;
+
+    public override void _Ready()
+    {
+        GD.Randomize();
+        _fruitsFull = _fruits.Duplicate();
+        _fruits.Shuffle();
+
+        for (int i = 0; i < 100; i++)
+        {
+            GD.Print(GetFruit());
+        }
+    }
+
+    public string GetFruit()
+    {
+        if(_fruits.Count == 0)
+        {
+            // Fill the fruits array again and shuffle it.
+            _fruits = _fruitsFull.Duplicate();
+            _fruits.Shuffle();
```

Comment:
If we use System collections, shuffling can be implemented as a custom method:

```csharp
private void Shuffle(Metal[] metals)
{
	for (int i = 0; i < metals.Length - 1; i++)
	{
		int j = GD.RandRange(i, metals.Length);
		if (j != i)
		{
			(metals[i], metals[j]) = (metals[j], metals[i]);
		}
	}
}
```

@paulloz Do you think this is good enough or would it be better to keep using Godot collections for the convenience of the `Shuffle` method?

Reply:
It really depends on what we want to emphasize on this page. Unless we explicitly want to emphasize usage of `Godot.Collections`, I'd rather use system ones.  

On a somewhat unrelated note, we could implement a `GD.Shuffle()` (or somewhere else in the API) that uses the Godot random, and list it on the [API differences](https://docs.godotengine.org/en/stable/tutorials/scripting/c_sharp/c_sharp_differences.html) page at some point.

Reply:
> I also think that any valid version of this is better than what we currently have

Agreed, so let's not block the PR on this. I'll resolve.